src/InputManager.hpp

```diff
@@ -23,8 +23,29 @@ namespace SokuLib
 	//CInputManagerCluster
 	struct KeymapManager {
 		char unknown[0x4];
+		// isPlayer is -1 when using keyboard, 0 when using first gamepad, 1 when using second gamepad.
 		char isPlayer;
-		char unknown2[0x33];
+		char unknown2[0x3];
+		union
```

Comment:
I don't think the union is really worth it here, I think the "bindings" field is that necessary. Removing the struct + union would make it more readable probably. You can move the struct out if you think having "binding" in front of everything is weird, or make it a int[], indexed using an enum.